wire-schema/src/main/java/com/squareup/wire/schema/EnumType.java

```diff
@@ -13,156 +13,125 @@
 
-  /** Returns the constant named {@code name}, or null if this enum has no such constant. */
-  public EnumConstant constant(String name) {
-    for (EnumConstant constant : constants()) {
-      if (constant.name().equals(name)) {
-        return constant;
-      }
-    }
-    return null;
-  }
-
-  /** Returns the constant tagged {@code tag}, or null if this enum has no such constant. */
-  public EnumConstant constant(int tag) {
-    for (EnumConstant constant : constants()) {
-      if (constant.tag() == tag) {
-        return constant;
-      }
-    }
-    return null;
-  }
+  /** Returns the constant tagged `tag`, or null if this enum has no such constant.  */
+  fun constant(tag: Int) = constants.find { it.tag() == tag }
 
-  public ImmutableList<EnumConstant> constants() {
-    return constants;
-  }
+  fun constants() = constants
 
-  @Override void link(Linker linker) {
-  }
+  internal override fun link(linker: Linker) {}
 
-  @Override void linkOptions(Linker linker) {
-    options.link(linker);
-    for (EnumConstant constant : constants) {
-      constant.linkOptions(linker);
+  internal override fun linkOptions(linker: Linker) {
+    options.link(linker)
+    for (constant in constants) {
+      constant.linkOptions(linker)
     }
-    allowAlias = options.get(ALLOW_ALIAS);
+    allowAlias = options.get(ALLOW_ALIAS)
   }
 
-  @Override void validate(Linker linker) {
-    linker = linker.withContext(this);
+  internal override fun validate(linker: Linker) {
+    var linker = linker
+    linker = linker.withContext(this)
 
-    if (!"true".equals(allowAlias)) {
-      validateTagUniqueness(linker);
+    if ("true" != allowAlias) {
+      validateTagUniqueness(linker)
     }
   }
 
-  private void validateTagUniqueness(Linker linker) {
-    Multimap<Integer, EnumConstant> tagToConstant = LinkedHashMultimap.create();
-    for (EnumConstant constant : constants) {
-      tagToConstant.put(constant.tag(), constant);
+  private fun validateTagUniqueness(linker: Linker) {
+    val tagToConstants = linkedMapOf<Int, MutableList<EnumConstant>>()
+    constants.forEach {
+      tagToConstants
+          .getOrPut(it.tag()) { mutableListOf() }
+          .add(it)
     }
 
-    for (Map.Entry<Integer, Collection<EnumConstant>> entry : tagToConstant.asMap().entrySet()) {
-      if (entry.getValue().size() > 1) {
-        StringBuilder error = new StringBuilder();
-        error.append(String.format("multiple enum constants share tag %s:", entry.getKey()));
-        int index = 1;
-        for (EnumConstant constant : entry.getValue()) {
-          error.append(String.format("\n  %s. %s (%s)",
-              index++, constant.name(), constant.location()));
+    for ((tag, constants) in tagToConstants) {
+      if (constants.size > 1) {
+        val error = buildString {
+          append("multiple enum constants share tag $tag:")
+          var index = 1
+          constants.forEach {
```

Comment:
`constants.forEachIndexed { index, constant ->` to avoid manually incrementing `index`